src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -578,7 +581,10 @@ def select(self, cols: List[Expression]) -> "SelectStatement":
             new._projection_in_str = self._projection_in_str
             new._schema_query = self._schema_query
             new.column_states = self.column_states
-            new._snowflake_plan = self._snowflake_plan
+            new._snowflake_plan = (
+                None  # The new._snowflake_plan will be populated when it's None.
```

Comment:
can you add one more comment that why we couldn't assign `self._snowflake_plan` directly here?

Reply:
Added more comments.